src/components/common/NetworkSetter.vue

```diff
@@ -35,9 +30,13 @@ export default {
       /* istanbul ignore next */
       update(data) {
         const networks = data.networks.map(network => network.id)
-        if (this.$route.params.networkId && this.findNetwork(networks)) {
+        let networkTitle
+        if (
+          this.$route.params.networkId &&
+          (networkTitle = this.findNetwork(networks))
```

Comment:
```suggestion
          (networkTitle === this.findNetwork(networks))
```

Reply:
Ok, this actually works. I am learning so much looking at your code @iambeone !

Reply:
this might work, but is confusing, this assignes networkTitle and returns true if networkTitle is not undefined. Let's make the code more readable by using findNetwork in line 33 and using a this.$route.params.networkId check inside of findNetwork